modin/distributed/dataframe/pandas/partitions.py

```diff
@@ -50,23 +50,24 @@ def unwrap_partitions(api_layer_object, axis=None, get_ip=False):
             f"Only API Layer objects may be passed in here, got {type(api_layer_object)} instead."
         )
 
+    modin_frame = api_layer_object._query_compiler._modin_frame
     if axis is None:
 
         def _unwrap_partitions():
+            modin_frame._propagate_index_objs(None)
+            [p.drain_call_queue() for p in modin_frame._partitions.flatten()]
```

Comment:
Shouldn't we drain call queue before propagating index objs? (trying to future proof as we begin to explore more lazy eval)

Reply:
`_propagate_index_objs` [adds](https://github.com/modin-project/modin/blob/88c3c33cf9b8aab5f5b7077555bea9f2d8466534/modin/core/dataframe/pandas/dataframe/dataframe.py#L500-L544) the index updates to the call queue, so we should call `_propagate_index_objs` first.